.github/workflows/java_compatibility.yml

```diff
@@ -0,0 +1,30 @@
+name: Test Java Compatibility
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+jobs:
+  build:
+    strategy:
```

Comment:
FYI, this might help with making sure each of these always run: https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#handling-failures

Reply:
perfect